lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -481,7 +481,10 @@ bool oclIsBuiltin(StringRef Name, StringRef &DemangledName, bool IsCpp) {
   } else {
     size_t Start = Name.find_first_not_of("0123456789", 2);
     size_t Len = 0;
-    Name.substr(2, Start - 2).getAsInteger(10, Len);
+    if (Name.substr(2, Start - 2).getAsInteger(10, Len)) {
+      SPIRVDBG(errs() << "Error in extracting integer value");
+      return false;
+    }
```

Comment:
This is a micro-optimization that avoids an extra jump in scenarios where the condition is always true.
```suggestion
    if (!Name.substr(2, Start - 2).getAsInteger(10, Len)) {
        DemangledName = Name.substr(Start, Len);
        return true;
    }
    SPIRVDBG(errs() << "Error in extracting integer value");
    return false;
```
Additionally, is not using the return value from a function a real issue? According to the specifications of C and C++, it is allowed. From a logical standpoint, what problems could arise if we fall through this check

Reply:
If we fall through, we return 'false' for isOCLBuiltin check and I think the caller will handle this correctly.

Thanks

Reply:
Also, about the optimization, I was just trying to keep it similar to what is being done a few lines above. I do not have any issues either ways. Can you please take a look at the code above this change and let me know what you think?

Thanks

Reply:
I just suggest not to worsen already not great performance in favor of satisfaction of some static analyzer tool reporting not critical issues.

Reply:
Yeah, I should probably be more consistent in my reviews, and for some reasons didn't complain in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1778/